documentation/smg/docs_discussion.md

```diff
@@ -0,0 +1,203 @@
+   * HORACE_VERSION determines output folder, copies whole html docs to both HORACE_VERSION and stable
+* Repo is general write access
+
+I will dicuss each of these in turn. Note that some of these are not mutually exclusive and could be used in combination.
+
+## Docs Repo
+### Current state
+
+#### Separate Repo
+
+Basically, the way I see it is that the separate repo allows us more freedom to build in a way in which we don't have to concern ourselves with accidentally disturbing other parts of the Horace project at large. This will serve to replace the wiki as the main docs source, and it might even be best to replace the http://horace.isis.rl.ac.uk/ with a redirect to the docs stable, whatever that may be.
+
+##### Pros:
+* Separates build hooks from main PRs
+* Prevents accidental damage to main repo on deploys
+* Allows documentation of not just Horace, but Herbert without causing confusion if renamed to PACE-docs, etc.
+
+##### Cons:
+* Separation may confuse people who want offline docs
+* May discourage keeping docs up to date as they are in different project
+
+### Alternatives
+
+#### Unified repo 
+Merge horace-docs into Horace branch as docs folder.
+
+##### Pros:
+* Locality of code and docs
+* Encourage people to update as they code, rather than as an afterthought
+* Keeps docs always in-line with the main build commit
+
+##### Cons:
+* May confuse builds/deploys (Could be mitigated through docs labels)
+* Disallows combination of Herbert/Horace/Entire PACE docs into one place.
+
+#### Submodule
+Use git submodules to have the docs as a sub-repo of Horace main
+
+##### Pros:
+* Still have the separation of repos
+* Docs exist in Horace repo and can be updated/held there
+
+###### Cons:
+* Requires knowledge to manage submodules
+* May require changing build scripts
+* Means that the docs will be up-to-date with latest docs not current Horace commit
+
+## Repo Builds
+
+### Current state
+#### Manual trigger
+Currently a docs build/deploy is manually triggered from the Jenkins pipeline, which will build and (on build success) deploy to a github.io accessible place.
+
+##### Pros:
+* Control over when/where docs get built
+* Little risk of accidental destruction of docs site due to manual version specification
+
+##### Cons:
+* Requires manual effort to say when to deploy
+* Could lead to desynchronisation of repo and site if care not taken 
+
+### Alternatives
+#### Build/Deploy on updated main
+On every PR merge new build, replacing current docs
+
+##### Pros:
+* Keep docs up-to-date when they are updated
+* No effort to update docs
+
+##### Cons:
+* May lead to damage if docs not verified properly (mitigated by multiple docs versions)
+* Difficult to check how branches would appear
+* Requires manually updating version number in Jenkins
+
+#### Build/Deploy on Horace release
+Tie docs deploy into the existing deploy pipeline such that on each major Horace release docs are released to the right tag
```

Comment:
We use the deploy pipeline to create draft releases which developers can then manually test (see [the deployment SMG](https://github.com/pace-neutrons/Horace/blob/master/documentation/smg/12_deploying_releases_to_GitHub.md)). So deploying the docs in the same pipeline could result in the documentation for a release appearing before the actual release package.

I'm not sure what the correct solution here is, but we could set up a [GitHub webhook that pings our build servers when a release is published](https://developer.github.com/webhooks/event-payloads/#release), our servers can then deploy the docs. Although it maybe simpler for the release docs to go up before the packages.